### PR TITLE
Add linux-htb QoS and Ingress/Egress tests

### PR DESCRIPTION
**Pull request by mateoflorido**
## Changes
- Added tests for linux-htb performance and QoS change in queues.
- Added test for pod ingress/egress limits.

### CODE REVIEW

File: tests/integration/test_kube_ovn.py [hunk 1]
```diff
@@ -51,3 +60,155 @@ async def test_kubectl_ko_plugin(ops_test: OpsTest):
...
+    ops_test: OpsTest, kubeconfig: Path, kubernetes: Client
+):
+    os.environ["KUBECONFIG"] = str(kubeconfig)
+    resources = Path.cwd() / "tests/data/qos_daemonset.yaml"
+    cmd = f"kubectl apply -f {resources}"
+    rc, stdout, stderr = await ops_test.run(*shlex.split(cmd))
+    assert rc == 0, f"Failed to deploy resources: {(stderr or stdout).strip()}"
+
+    # Sleep to wait for resources to become available
+    time.sleep(60)
```
Comment (stonepreston) [inline tests/integration/test_kube_ovn.py:119]: lightkube client wait()

**Comment by mateoflorido** [comment]
Thanks for the feedback @stonepreston, last commit implemented all the suggested changes.

File: tests/integration/test_kube_ovn.py [hunk 2]
```diff
@@ -51,3 +60,180 @@ async def test_kubectl_ko_plugin(ops_test: OpsTest):
...
+    rate_values = (
+        'ovn.kubernetes.io/ingress_rate="10" ovn.kubernetes.io/egress_rate="5"'
+    )
+    cmd = (
+        f"kubectl --kubeconfig {kubeconfig} "
+        "annotate --overwrite "
+        f"pod {test_pod.metadata.name} "
+        f"-n {namespace} {rate_values}"
+    )
+    rc, stdout, stderr = await ops_test.run(*shlex.split(cmd))
```
Comment (addyess) [inline tests/integration/test_kube_ovn.py:95]: you can annotate with lightkube library.  

it seems annotations are a part of the metadata object of the pod.  But i get this is just so much faster. :)

**Review by addyess — APPROVED** [review]
Yeah, i see this as well reasoned.

**Review by stonepreston — APPROVED** [review]

**Pull request merged**